### PR TITLE
Update doc about json tag is needed to generate informer externalversion package

### PR DESCRIPTION
**Pull request by CatherineF-dev**
#### What type of PR is this?
/kind documentation

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/sample-controller/README.md [hunk 1]
```diff
@@ -126,6 +126,9 @@ type User struct {
 }
 ```
 
+Note, json tag `json:` is needed to generate informer externalversions package.
+Otherwise, informer internalversion package is generated.
```
Comment (JoelSpeed) [inline staging/src/k8s.io/sample-controller/README.md:130]: I think perhaps we want to be a little less specific here, while yes, the informer generation needs the JSON tags, there are many other generators that also need the tags. For example, applyconfiguration-gen, lister-gen and controller-gen also all treat a non json tagged member as internal, and therefore either create internal listers or omit the generated schemas.

Perhaps we want to go with something along the lines of 
```suggestion
Note, the JSON tag `json:` is required on all user facing fields within your type. Typically API types contain only user facing fields. When the JSON tag is omitted from the field, Kubernetes generators consider the field to be internal and will not expose the field in their generated external output. For example, this means that the field would not be included in a generated CRD schema.
```

On the informer `internalversions`/`externalversions` thing, it's my understanding that it only generates `internalversions` when it sees an internal field, and `externalversions` when it sees an external field, so I'm guessing when you ran into this, you had no json tags at all across your struct?

Comment (CatherineF-dev) [reply staging/src/k8s.io/sample-controller/README.md]: Thanks! Have merged your suggestions!

> so I'm guessing when you ran into this, you had no json tags at all across your struct?

Yes, correct. I found json tag is needed when looking at source codes of code-generator.

**Comment by JoelSpeed** [comment]
/lgtm

Will need someone from sig-api-machinery to take a look and check they agree

**Comment by deads2k** [comment]
I'd be happy to see the internal type generators for clients, informers, and listers be removed.  The internal client-side tooling only works when the client is fully aware of all possible defaulting and conversion. We decided several years ago that such things are only the domain of the server.  This PR improves the current state of the world, but I'd be happy to see another PR that removes the possibility of internal generation entirely.

/approve

**Comment by CatherineF-dev** [comment]
cc @munnerz @sttts  Could you help approve this small PR when you're available? Thanks! I found it needs approvals from OWNER.

**Pull request merged**